pkg/controllers/mcs/serviceexport_controller.go

```diff
@@ -0,0 +1,217 @@
+/*
+Copyright 2022 The Clusternet Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package mcs
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/dixudx/yacht"
+	discoveryv1 "k8s.io/api/discovery/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+	mcsclientset "sigs.k8s.io/mcs-api/pkg/client/clientset/versioned"
+	mcsInformers "sigs.k8s.io/mcs-api/pkg/client/informers/externalversions/apis/v1alpha1"
+
+	"github.com/clusternet/clusternet/pkg/known"
+	"github.com/clusternet/clusternet/pkg/utils"
+)
+
+func init() {
+	utilruntime.Must(v1alpha1.AddToScheme(scheme.Scheme))
+}
+
+type reconciler struct {
+	//local msc client
+	mcsClientset    *mcsclientset.Clientset
+	localk8sClient  kubernetes.Interface
+	parentk8sClient kubernetes.Interface
+	// child cluster dedicated namespace
+	dedicatedNamespace string
+}
+
+func NewSeController(parentDedicatedKubeConfig *rest.Config, localClient kubernetes.Interface, mcsClientset *mcsclientset.Clientset,
+	seInformers mcsInformers.ServiceExportInformer, namespace string, workers int) *yacht.Controller {
+	parentClient := kubernetes.NewForConfigOrDie(parentDedicatedKubeConfig)
+
+	r := &reconciler{
+		mcsClientset:       mcsClientset,
+		parentk8sClient:    parentClient,
+		localk8sClient:     localClient,
+		dedicatedNamespace: namespace,
+	}
+
+	c := yacht.NewController("serviceexport").WithWorkers(workers).WithCacheSynced(seInformers.Informer().HasSynced).WithHandlerFunc(r.Handle)
+	seInformers.Informer().AddEventHandler(c.DefaultResourceEventHandlerFuncs())
+
+	return c
+}
+
+func (r *reconciler) Handle(obj interface{}) (requeueAfter *time.Duration, err error) {
+	ctx := context.Background()
+	key := obj.(string)
+	namespace, seName, err := cache.SplitMetaNamespaceKey(key)
+	utilruntime.Must(err)
+
+	se, err := r.mcsClientset.MulticlusterV1alpha1().ServiceExports(namespace).Get(ctx, seName, metav1.GetOptions{})
```

Comment:
Shall we use `mcsLister` to `Get` a resource?

Reply:
ok, done